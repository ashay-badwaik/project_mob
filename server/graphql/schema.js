const { gql } = require('apollo-server');


const typeDefs = gql`
    type UserDetails{
        id: ID!
        name: String
        bio: String
        image_path: String
    }


    type Query {
        getAllUserDetails: [UserDetails]!
        getUserDetails(id:ID!): UserDetails!
    }

    type Mutation {
        addUser(name:String! bio:String!): String!
        uploadImage(id:ID! path:String!): String!
        uploadPath(id:ID! path:String!): String!
    }
`

module.exports = typeDefs;