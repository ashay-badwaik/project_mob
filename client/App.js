/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 *
 * @format
 * @flow strict-local
 */

import React from 'react';
import { NavigationContainer } from '@react-navigation/native';
import { createNativeStackNavigator } from '@react-navigation/native-stack';
import { ApolloClient, InMemoryCache, ApolloProvider } from '@apollo/client';

import { Text, View } from 'react-native';

import HomeScreen from './components/homeScreen';
import StoryScreen from './components/storyScreen';
import AddStoryScreen from './components/addStoryScreen';
import ChangeProfilePic from './components/changeProfilePic';


const client = new ApolloClient({
  uri: 'http://localhost:4000/',
  cache: new InMemoryCache(),
})

const Stack = createNativeStackNavigator();

const App = () => {
  return (
    <ApolloProvider client={client}>
      <NavigationContainer>
        <Stack.Navigator initialRouteName="Home">
          <Stack.Screen name="Home" component={HomeScreen} options={{headerShown: false}}/>
          <Stack.Screen name="Story" component={StoryScreen}/>
          <Stack.Screen name="AddStory" component={AddStoryScreen} />
          <Stack.Screen name="ChangeProfilePic" component={ChangeProfilePic} />
        </Stack.Navigator>
      </NavigationContainer>
    </ApolloProvider>
  );
}

export default App;project_mov
