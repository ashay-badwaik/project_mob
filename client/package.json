{
  "name": "client",
  "version": "0.0.1",
  "private": true,
  "scripts": {
    "android": "react-native run-android",
    "ios": "react-native run-ios",
    "start": "react-native start",
    "test": "jest",
    "lint": "eslint ."
  },
  "dependencies": {
    "@apollo/client": "^3.4.7",
    "@react-navigation/native": "^6.0.1",
    "@react-navigation/native-stack": "^6.0.2",
    "graphql": "^15.5.1",
    "react": "17.0.1",
    "react-native": "^0.64.2",
    "react-native-image-crop-picker": "^0.36.2",
    "react-native-safe-area-context": "^3.2.0",
    "react-native-screens": "^3.5.0",
    "react-native-svg": "^9.13.6",
    "react-native-svg-transformer": "^0.14.3"
  },
  "devDependencies": {
    "@babel/core": "7.15.0",
    "@babel/runtime": "7.14.8",
    "@react-native-community/eslint-config": "2.0.0",
    "babel-jest": "26.6.3",
    "eslint": "7.14.0",
    "jest": "26.6.3",
    "metro-react-native-babel-preset": "0.64.0",
    "react-test-renderer": "17.0.1"
  },
  "jest": {
    "preset": "react-native"
  }
}
