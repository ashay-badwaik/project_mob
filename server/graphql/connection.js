const mysql = require('mysql2');

const connection = mysql.createConnection({
    host: 'localhost',
    port: 3306, 
    user: 'root',
    password: '@vengers',
    database: 'story'
});

module.exports = connection;