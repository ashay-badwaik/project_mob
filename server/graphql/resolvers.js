const mysql = require('mysql2');
const fs = require("fs");
const connection = require("./connection");


const Query = {
    getAllUserDetails: async () => {
        let QUERY = `
            SELECT id, name, bio, image_path 
            FROM users
        `
        // console.log(QUERY);
        try {
            let data = await connection.promise().query(QUERY);
            // console.log(data[0]);
            return data[0];
        } catch (err) {
            console.log(err);
        }
    },

    getUserDetails: async (_, { id }) => {
        let QUERY = `
            SELECT id, name, bio, image_path 
            FROM users 
            WHERE id=${id}
        `
        try {
            let data = await connection.promise().query(QUERY);
            // console.log(data[0][0]);
            return data[0][0];
        } catch (err) {
            console.log(err);
        }
    }
}


const Mutation = {
    addUser: async (_, { name, bio }) => {
        let QUERY = `
            INSERT INTO users (name, bio) 
            VALUES ("${name}", "${bio}")
        `

        try {
            const success = await connection.promise().query(QUERY);
            if (success) return "Success";
        } catch (err) {
            console.log(err);
        }
    },

    uploadImage: async (_, { id, path }) => {
        let QUERY = `
            UPDATE trial 
            SET picture=LOAD_FILE('${path}') 
            WHERE id=${id}
        `

        try {
            const success = await connection.promise().query(QUERY);
            if (success) return "image uploaded";
        } catch (err) {
            console.log(err);
        }
    },

    uploadPath: async (_, { id, path }) => {
        let QUERY = `
            UPDATE users 
            SET image_path='${path}' 
            WHERE id=${id}
        `

        try {
            const success = await connection.promise().query(QUERY);
            if (success) return "image path uploaded";
        } catch (err) {
            console.log(err);
        }
    }
}

module.exports = { Query, Mutation }; 