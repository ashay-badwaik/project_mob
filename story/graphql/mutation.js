import { gql } from "@apollo/client";


export const UPLOAD_IMAGE_PATH = gql`
    mutation uploadPath($id: ID!, $path: String!) {
        uploadPath(
            id: $id 
            path: $path
        )
    }
`