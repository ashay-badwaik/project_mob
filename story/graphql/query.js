import { gql } from '@apollo/client';

export const USER_DETAILS = gql`
  query getUserDetails($id: ID!){
    getUserDetails(id:$id){
      name
      bio
      imagePath
    }
  }
`