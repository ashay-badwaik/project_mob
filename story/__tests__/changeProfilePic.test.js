import 'react-native';
import React from 'react';
import { MockedProvider } from "@apollo/client/testing";
import ChangeProfilePic from './../components/changeProfilePic';
import ImagePicker from 'react-native-image-crop-picker';

// Note: test renderer must be required after react-native.
import renderer, { act } from 'react-test-renderer';
import { UPLOAD_IMAGE_PATH } from '../graphql/mutation';

jest.useFakeTimers()

const mockedParams = {
    route: {
        params: {
            imagePath: './../images/no_profile_pic.jpeg'
        }
    },
    navigation: {
        navigate: jest.fn()
    }
}

const mock = [
    {
        request:{
            query: UPLOAD_IMAGE_PATH,
            variable: {
                id: 1,
                path: './abs/sks.jpeg'
            }
        },
        result: {
            data: "image path uploaded"
        } 
    }
]



it('renders correctly', () => {
    const tree = renderer.create(
        <MockedProvider addTypename={false} mocks={mock}>
            <ChangeProfilePic {...mockedParams}/>
        </MockedProvider>
    )
})

it('snapshot', () => {
    const tree = renderer.create(
        <MockedProvider addTypename={false} mocks={mock}>
            <ChangeProfilePic {...mockedParams}/>
        </MockedProvider>
    )
    expect(tree).toMatchSnapshot()
})

describe('buttons', () => {
    const tree = renderer.create(
        <MockedProvider addTypename={false} mocks={mock}>
            <ChangeProfilePic {...mockedParams}/>
        </MockedProvider>
    )

    it('gallery', () => {
        const btn = tree.root.findByProps({testID: 'gallery'}).props;
        btn.onPress();

        expect(ImagePicker.openPicker).toHaveBeenCalled()
    })

    it('camera', () => {
        const btn = tree.root.findByProps({testID: 'camera'}).props;
        btn.onPress();

        expect(ImagePicker.openCamera).toHaveBeenCalled()
    })
    it('upload', () => {
        const btn = tree.root.findByProps({testID: 'upload'}).props;
        btn.onPress();

        expect(mockedParams.navigation.navigate).toHaveBeenCalledWith(expect.objectContaining({name:'Home'}))
    })
})


it('image display', async () => {
    const tree = renderer.create(
        <MockedProvider addTypename={false} mocks={mock}>
            <ChangeProfilePic {...mockedParams}/>
        </MockedProvider>
    )

    const btn = tree.root.findByProps({testID:'gallery'}).props;
    await act( async () => {
        await btn.onPress();
    })

    const img = tree.root.findByProps({testID:'picture'});
    expect(img).not.toBeNull()
})

