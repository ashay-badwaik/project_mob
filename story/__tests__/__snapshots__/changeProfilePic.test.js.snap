// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "height": "100%",
    }
  }
>
  <Text
    style={
      Object {
        "color": "#fdbb21",
        "fontSize": 30,
        "marginHorizontal": 30,
        "marginVertical": 15,
        "textAlign": "center",
      }
    }
  >
    Set Profile Pic
  </Text>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderColor": "#fdbb21",
        "borderRadius": 30,
        "borderWidth": 2,
        "height": 300,
        "justifyContent": "center",
        "marginHorizontal": 30,
        "marginVertical": 15,
        "width": 300,
      }
    }
    testID="image"
  >
    <Image
      source={
        Object {
          "uri": "./../images/no_profile_pic.jpeg",
        }
      }
      style={
        Object {
          "borderRadius": 30,
          "height": "100%",
          "width": "100%",
        }
      }
      testID="picture"
    />
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fdbb21",
        "borderRadius": 20,
        "height": 60,
        "justifyContent": "center",
        "marginHorizontal": 30,
        "marginVertical": 15,
        "opacity": 1,
        "width": 300,
      }
    }
    testID="gallery"
  >
    <Text
      style={
        Object {
          "color": "#fff",
          "fontSize": 30,
          "lineHeight": 30,
        }
      }
    >
      Pick from Gallery
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fdbb21",
        "borderRadius": 20,
        "height": 60,
        "justifyContent": "center",
        "marginHorizontal": 30,
        "marginVertical": 15,
        "opacity": 1,
        "width": 300,
      }
    }
    testID="camera"
  >
    <Text
      style={
        Object {
          "color": "#fff",
          "fontSize": 30,
          "lineHeight": 30,
        }
      }
    >
      Take a photo
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fdbb21",
        "borderRadius": 20,
        "height": 60,
        "justifyContent": "center",
        "marginHorizontal": 30,
        "marginVertical": 15,
        "opacity": 1,
        "width": 300,
      }
    }
    testID="upload"
  >
    <Text
      style={
        Object {
          "color": "#fff",
          "fontSize": 30,
          "lineHeight": 30,
        }
      }
    >
      Upload
    </Text>
  </View>
</View>
`;
